priv/post.ex.eex

```diff
@@ -30,22 +29,21 @@ defmodule <%= context.module_name %> do
     ]
 <%= if context.protocol == "query" do %>
     input = Map.merge(input, %{"Action" => action, "Version" => "<%= context.api_version %>"})<% end %>
-    payload = <%= context.encode %>
+    payload = encode!(client, input)
     headers = AWS.Request.sign_v4(client, "POST", url, headers, payload)
+    post(client, url, payload, headers, options)
+  end
 
-    case HTTPoison.post(url, payload, headers, options) do
-      {:ok, %HTTPoison.Response{status_code: 200, body: ""} = response} ->
-        {:ok, nil, response}
-
-      {:ok, %HTTPoison.Response{status_code: 200, body: body} = response} ->
-        {:ok, <%= context.decode %>, response}
+  defp post(client, url, payload, headers, options) do
+    case do_request(client, :post, url, payload, headers, options) do
+      {:ok, %{status_code: 200, body: body} = response} ->
+        body = if(body != "", do: decode!(client, body))
```

Comment:
this is minor, but here you don't need the parenthesis on `if`:

```suggestion
        body = if body != "", do: decode!(client, body)
```

Reply:
I didn't know that. Thank you!